### PR TITLE
Implement message logging via a new "Observer" pipeline stage.

### LINKED ISSUE #4: Add inbound/outbound pipelines for logging messages

### PR DESCRIPTION
**Pull request by jmalloc**
Fixes #4

### CODE REVIEW

**Comment by jmalloc** [comment]
This PR still has some skipped tests.

**Comment by jmalloc** [comment]
Ok, ready for @koden-km eyes.

File: src/ax/observability/observer.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// Observer is an interface for types that implement one or more of the more
+// specific observer interfaces.
+type Observer interface {
+}
+
+// BeforeInboundObserver is an interface for observers that are notified before
+// the inbound pipeline accepts a message.
+type BeforeInboundObserver interface {
+	// BeforeInbound is called before a message is passed to the next pipeline stage.
+	// The returned context replaces ctx in calls to other observers and the next pipeline stage.
```
Comment (koden-km) [inline src/ax/observability/observer.go]: As mentioned in chat, there is no returned context.

File: src/ax/observability/observer.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	// AfterInbound is called after a message is accepted by the next pipeline stage.
+	// err is the error returned by the next pipeline stage, which may be nil.
+	AfterInbound(ctx context.Context, env bus.InboundEnvelope, err error)
+}
+
+// BeforeOutboundObserver is an interface for observers that are notified before
+// the outbound pipeline accepts a message.
+type BeforeOutboundObserver interface {
+	// BeforeOutbound is called before a message is passed to the next pipeline stage.
+	// The returned context replaces ctx in calls to other observers and the next pipeline stage.
```
Comment (koden-km) [inline src/ax/observability/observer.go]: As mentioned in chat, there is no returned context.

File: src/ax/message_test.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package ax_test
```
Comment (koden-km) [inline src/ax/message_test.go]: Is this file supposed to be all commented out?

Comment (jmalloc) [reply src/ax/message_test.go]: It's not supposed to exist at all!

**Pull request merged**